solc/CommandLineInterface.cpp

```diff
@@ -584,6 +587,37 @@ bool CommandLineInterface::parseLibraryOption(string const& _input)
 	return true;
 }
 
+//read file contents from m_sourceCodes into m_sourceJsons, which either are
+// a JSON-file as produced by
+// a) --combined-json -ast,compact-format
+// b) the standard-json output
+map<string, Json::Value const*> CommandLineInterface::parseAstFromInput()
+{
+	map<string, Json::Value const*> sourceJsons;
+	map<string, string> tmp_sources; //used to generate the onchainmetadata-hash
+	for (auto const& srcPair: m_sourceCodes)
+	{
+		Json::Value* ast = new Json::Value();
+		// try parsing as json
+		if (jsonParse(srcPair.second, *ast))
+			if (ast->isMember("sources"))
```

Comment:
Why do we have a list of sources here and also iterate over the input sources `m_sourceCodes`? What happens if there are conflicts (e.g. multiple files of the same name)?

Reply:
hah, good catch. added an assertion that all sourcenames must be unique